test/monitoringlocationlinter/analyzer.go

```diff
@@ -0,0 +1,48 @@
+package monitoringlocationlinter
+
+import (
+	"go/ast"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+
+	"github.com/monitoring/test/monitoringlocationlinter/prometheushandler"
+)
+
+// run is the main assertion function
+func run(pass *analysis.Pass) (interface{}, error) {
+	for _, file := range pass.Files {
+		if shouldLintFile(pass.Fset.File(file.Pos()).Name()) {
+			prometheusHandler := prometheushandler.GetPrometheusHandler(file)
+			if prometheusHandler == nil { // no prometheus import => no use in prometheus in this file; nothing to do here
+				continue
+			}
+
+			ast.Inspect(file, func(node ast.Node) bool {
+				call, ok := node.(*ast.CallExpr)
+				if ok && prometheusHandler.IsRegisterMetricsCall(call) {
+					pass.Reportf(node.Pos(), "monitoring-location-linter: Prometheus metrics should be registered only under pkg/monitoring.")
+					return true
+				}
+
+				return true
+			})
```

Comment:
nit: I think you can move this to a function named `lintFile(file)`

Reply:
Done